src/custom/components/AccountDetails/Transaction.tsx

```diff
@@ -63,22 +83,108 @@ const TransactionState = styled(OldTransactionState).attrs(
   ${(props): string | false => !!props.disableMouseActions && `pointer-events: none; cursor: none;`}
 `
 
+type CancellationModalProps = {
+  onDismiss: () => void
+  isOpen: boolean
+  orderId: string
+  summary: string | undefined
+}
+
+function CancellationModal(props: CancellationModalProps): JSX.Element | null {
+  const { orderId, isOpen, onDismiss, summary } = props
+  const shortId = shortenOrderId(orderId)
+
+  const [status, setStatus] = useState<'not started' | 'waiting for wallet' | 'error'>('not started')
+  const [error, setError] = useState('')
+  const [showMore, setShowMore] = useState(false)
+  const cancelOrder = useCancelOrder()
+
+  const toggleShowMore = () => setShowMore(showMore => !showMore)
+
+  useEffect(() => {
+    // Reset status every time orderId changes to avoid race conditions
+    setStatus('not started')
+  }, [orderId])
+
+  const onClick = useCallback(() => {
+    setStatus('waiting for wallet')
+    setError('')
+    cancelOrder(orderId)
+      .then(onDismiss)
+      .catch(e => {
+        setError(e.message)
+        setStatus('error')
```

Comment:
why do u need both status for error and an error? doesn't seem like u use that status apart from boolean checking if there should be an error displayed. could make error null and check like that

Reply:
Makes sense, will update